src/PointwiseFunctions/Hydro/EquationsOfState/HybridEos.cpp

```diff
@@ -0,0 +1,153 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/Hydro/EquationsOfState/HybridEos.hpp"
+
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+namespace EquationsOfState {
+template <typename ColdEquationOfState>
+HybridEos<ColdEquationOfState>::HybridEos(
+    ColdEquationOfState cold_eos, const double thermal_adiabatic_index) noexcept
+    : cold_eos_(std::move(cold_eos)),
+      thermal_adiabatic_index_(thermal_adiabatic_index) {}
+
+EQUATION_OF_STATE_MEMBER_DEFINITIONS(template <typename ColdEquationOfState>,
+                                     HybridEos<ColdEquationOfState>, double, 2)
+EQUATION_OF_STATE_MEMBER_DEFINITIONS(template <typename ColdEquationOfState>,
+                                     HybridEos<ColdEquationOfState>, DataVector,
+                                     2)
+
+template <typename ColdEquationOfState>
+HybridEos<ColdEquationOfState>::HybridEos(CkMigrateMessage*
+                                          /*unused*/) noexcept {}
```

Comment:
I'm not sure if the EOSes are set up for this, but can this call `EquationOfState<is_relativistic, 2>::EquationOfState(msg)`? Basically, according to the Charm++ manual we should be call the base class constructors in each of these